apps/server/src/modules/authorization/api/dto/authorization-body.params.ts

```diff
@@ -0,0 +1,44 @@
+import { Permission } from '@shared/domain/interface';
+import { ApiProperty } from '@nestjs/swagger';
+import { IsArray, IsEnum, IsMongoId, ValidateNested } from 'class-validator';
+import { Action, AuthorizableReferenceType, AuthorizationContext } from '../../domain';
+
+class AuthorizationContextParams implements AuthorizationContext {
+	@IsEnum(Action)
+	@ApiProperty({
+		description: 'Define for which action the operation should be performend.',
+		enum: Action,
+		example: Action.read,
+	})
+	action!: Action;
+
+	@IsArray()
+	@IsEnum(Permission, { each: true })
+	@ApiProperty({
+		enum: Permission,
+		isArray: true,
+		description: 'Needed user permissions based on user role, that are needed to execute the operation.',
+		example: Permission.USER_UPDATE,
+	})
+	requiredPermissions!: Permission[];
+}
+
+export class AuthorizationBodyParams {
+	@ValidateNested({ each: true })
```

Comment:
In the docs it says `{ each: true }` is only for array-like things: https://github.com/typestack/class-validator?tab=readme-ov-file#validating-arrays I don't think it makes sense here.

Reply:
Das zu fixen hat mich 50% der Änderungszeit gekostet. Was genau im package passiert und warum kann ich aber noch nicht sagen. War erstmal froh das ich eine funktionierende Lösung gefunden habe.